src/term/mod.rs

```diff
@@ -1818,6 +1816,12 @@ impl ansi::Handler for Term {
         self.colors = self.original_colors;
         self.color_modified = [false; color::COUNT];
         self.cursor_style = None;
+        self.grid = Grid::new(
```

Comment:
Rather than creating a new Grid and thus reallocating everything, we could instead clear the visible area and truncate available scrollback.